plugins/paymethod/manual/ManualPaymentPlugin.php

```diff
@@ -65,6 +65,7 @@ public function register($category, $path, $mainContextId = null)
             Hook::add('Form::config::before', [$this, 'addSettings']);
             Hook::add('Mailer::Mailables', [$this, 'addMailable']);
             Hook::add('Installer::postInstall', [$this, 'updateSchema']);
+            Hook::add('API::payments::settings::edit', [$this, 'updatePaymentSettings']);
```

Comment:
`PaymethodPlugin::register()` is called, so there will be multiple callbacks added to this hook for each payment plugin. For example, if I have the `ManualPaymentPlugin` and the `PaypalPaymentPlugin` enabled, the `API::payments::settings::edit` hook will call the following callbacks:

- `ManualPaymentPlugin::updatePaymentSettings()`
- `ManualPaymentPlugin::updatePaymentSettings()`
- `PaypalPaymentPlugin::updatePaymentSettings()`
- `PaypalPaymentPlugin::updatePaymentSettings()`

That's because when `parent::register()` is called, `[$this, 'updatePaymentSettings']` will refer to the current plugin as `$this`.

There are two ways to handle this:

1. You can remove the hook from the child plugins and rely on the hook in the parent plugin. If you do this, make `PaymethodPlugin::updatePaymentSettings()` an abstract function so that child classes have to implement it.
2. You can abandon the use of hooks to solve this problem, since it introduces a layer of complexity in the hook registration stuff that might not be necessary. I also don't love introducing two overlapping methods that sound like they do the same thing (`saveSettings`, `updatePaymentSettings`). If you go with this route, I think it's ok to change the function signature of `saveSettings` in a way that breaks compatibility. If you introduce type hints and make it an abstract function on the base class (`PaymethodPlugin`), any plugin that needs to be updated will fail early and with a very clear error message. This is one way we can lean on script typing to help third-party developers rather than rely on us documenting every single change.

I'm fine with either approach you choose. :+1: